e2e/optimism-stack.Dockerfile

```diff
@@ -0,0 +1,60 @@
+# Copyright (c) 2024 Hemi Labs, Inc.
+# Use of this source code is governed by the MIT License,
+# which can be found in the LICENSE file.
+
+FROM golang:1.22
+
+RUN apt-get update
+
+RUN apt-get install -y nodejs npm jq
+
+RUN npm install -g pnpm
+
+WORKDIR /git
+
+RUN curl -L https://foundry.paradigm.xyz | bash
+
+RUN . /root/.bashrc
+
+ENV PATH="${PATH}:/root/.foundry/bin"
+
+RUN foundryup
+
+ARG OP_GETH_CACHE_BREAK=1
+
+RUN git clone https://github.com/hemilabs/op-geth
+WORKDIR /git/op-geth
+
+# remove this before merge
+RUN git checkout clayton/local-eth
+
+RUN make
+RUN go install ./...
+RUN abigen --version
+
+ARG OPTIMISM_CACHE_BREAK=1
+
+WORKDIR /git
+RUN git clone https://github.com/hemilabs/optimism
+WORKDIR /git/optimism
+
+# remove this before merge
+RUN git checkout clayton/local-eth
+
+RUN git submodule update --init --recursive
+RUN pnpm install
+WORKDIR /git/optimism/packages/contracts-bedrock
+RUN sed -e '/build_info/d' -i ./foundry.toml
+WORKDIR /git/optimism
+RUN make op-bindings op-node op-batcher op-proposer
+RUN pnpm build
+
+WORKDIR /git/optimism/packages/contracts-bedrock
+RUN forge install
+RUN forge build
+
+WORKDIR /git/optimism
+
+# this is not great, but this command generates the needed files for devnet
+# and runs a failing test, ignore the failure but use the files
+RUN make devnet-allocs || :
```

Comment:
this is causing issues; need better solution

Reply:
You could wrap it in a script that runs the make command, expects the failure, checks for the generated files, does any cleanup, then reports success.

Reply:
That's a great idea. That would make this PR much smaller too since I wouldn't need to store the generates files in version controll. Will update 👍

Reply:
I'm just now remembering why I can't use this approach; I'll add some more detail in the AM

Reply:
hey @jcvernaleo ok so here is the pickle I am in with this:

with our current `hemi` branch of `optimism`, when running the `make devnet-allocs` command, I run into this issue:
```
/go/pkg/mod/github.com/cockroachdb/pebble@v0.0.0-20230928194634-aa077af62593/internal/cache/robin_hood.go:73:2: type [2147483647]byte larger than address space
```

this issue comes before the needed files are generated.  it's easily fixable, but it requires me to update `optimism` and `op-geth`, which is not ideal at this moment.

so, annoyingly, I think the safest thing to do here is generate the files from the updated `optimism` and `op-geth` branches, include them in version control, and not run the file generation command.  this means that we don't have to do anything for `optimism` and `op-geth` for this PR as well